pkg/workflows/testutils/runner.go

```diff
@@ -34,16 +56,69 @@ func (r *Runner) Run(factory *workflows.WorkflowSpecFactory) error {
 		return err
 	}
 
-	// TODO https://smartcontract-it.atlassian.net/browse/KS-442, implement this function
-	_ = spec
-	return errors.New("TODO https://smartcontract-it.atlassian.net/browse/KS-442")
+	if err = r.ensureGraph(spec); err != nil {
+		return err
+	}
+
+	r.setupSteps(factory, spec)
+
+	return r.walk(spec, workflows.KeywordTrigger)
```

Comment:
so a workflow always has a single trigger? is that set to change eventually?

Reply:
Yes, and most likely.

Reply:
Note a workflow can have multiple triggers, but it will only have a single trigger event (and therefore only a single "trigger" is the root for a given execution), which is why we can make this leap.

Lmk if that distinction doesn't make sense